### PR TITLE
Added a few variables to Makefile to simplify automated packaging on linux

### PR DESCRIPTION
**Pull request by luigino**
(2nd attempt)

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -42,15 +45,23 @@ else
...
+		LUA_INCLUDE ?= $(INCROOT)/include/lua5.2
+		LUA_LIB ?= -llua5.2
 		TESTER = bin/wordgrinder-debug
 	endif
 
-	NCURSES_CFLAGS := $(shell pkg-config ncursesw --cflags)
-	NCURSES_LIB := $(shell pkg-config ncursesw --libs)
-	X11_CFLAGS := $(shell pkg-config freetype2 --cflags) -I/usr/include/X11
-	X11_LIB := -lX11 -lXft $(shell pkg-config freetype2 --libs) 
+	ifndef NCURSES_CFLAGS
```
Comment (davidgiven) [inline Makefile]: Any reason not to use ?= here?

**Comment by davidgiven** [comment]
Thank you for that --- LGTM. One comment (hopefully inline?).

Would like to know why Travis didn't try to build this PR, though. It did for your other one (and it worked).

File: Makefile (hunk 1 above)
Comment (luigino) [inline Makefile]: I think with ?= everything gets expanded and the pkg-config command is executed anyway.

Comment (davidgiven) [inline Makefile]: Oh, yeah! I need a := variable to force evaluation right then.

(I _loathe_ make.)

Thanks!

**Pull request merged**